zappa/handler.py

```diff
@@ -583,7 +583,17 @@ def handler(self, event, context):
                         )
 
                     if response.data:
-                        if (
+                        if settings.BINARY_SUPPORT and response.headers.get(
+                            "Content-Encoding"
```

Comment:
The test case docstring says:

> With Binary Support enabled, response mimetypes that are not text/* or application/json will be base64 encoded

But wouldn't this base64 encode all responses so long as they have a `Content-Encoding` header (even if its mimetype is `text/*` or `application/json`)?  Perhaps I'm misunderstanding, but the logic here doesn't seem to match that docstring.

Reply:
Your understanding was correct. I was misreading the original PR. As written, the code will encode to b64 if BINARY_SUPPORT is enabled and a `Content-Encoding` header is present. If `Content-Encoding` isn't present, then it may still b64 encode under some more specific conditions.

While breaking up the tests to give categories of behavior it's own function, I changed some of the test documentation language to reflect this as well (https://github.com/zappa/Zappa/pull/971/commits/f096f4c16f367e6c439ce0bc47a9f58287a05eaa#diff-3f86b181ab338949acfc1d27ff975b666acd3f0b016eb4bcce7c79fab09e463bR230-R233).

Reply:
Your understanding is correct. The docstring text for the test is misleading. Good catch!

I've updated the tests and the language in [`eb2b50f` (#971)](https://github.com/zappa/Zappa/pull/971/commits/eb2b50f65acb1b04d2a3f3d74c5de6ae919c7fe1).